mesonbuild/options.py

```diff
@@ -473,8 +474,65 @@ def add_to_argparse(self, name: str, parser: argparse.ArgumentParser, help_suffi
     OptionKey('purelibdir', module='python'): {},
 }
 
-
 class OptionStore:
     def __init__(self):
-        # This class will hold all options for a given build directory
-        self.dummy = None
+        self.d: T.Dict['OptionKey', 'UserOption[T.Any]'] = {}
+
+    def __len__(self):
+        return len(self.d)
+
+    def ensure_key(self, key: T.Union[OptionKey, str]) -> OptionKey:
```

Comment:
Suggestion: this method could accept same kwargs as OptionKey, and forward them to the OptionKey initializer. That way, the OptionStore getters and setters could accept kwargs and you could do `get_value('key', machine='m')`, etc. instead of `get_value(OptionKey('key', machine='m'))`

Reply:
I considered that but chose against it, because then you have to deal with the case of `get_value(some_key_object, machine='m'))`. What if the machine in the key object is different from the one given as a kwarg? That is unexpected. You need to add validation code to error out if that happens.

The current approach is simpler: either you define _all_ of the optional values or _none_ of them.

Reply:
But you could do something like:
```
if isinstance(key, str):
    return OptionKey(key, **kwargs)
if kwargs:
    return key.evolve(**kwargs)
return key
```

Reply:
I could, but choose not to.

With that you can not tell what the method is actually doing without reading its contents. That is not good design.